### DISCUSSION TITLE
Why @babel/plugin-proposal-decorators implements is different from tc39/proposal-decorators?

### DISCUSSION THREAD

**Discussion by XGHeaven** [babel/babel#13662] [General]
I found babel implements decorator signature is:
```ts
type Decorator = (description: DecoratorDescription) => DecoratorDescription
interface DecoratorDescription {
  // xxx
}
```

However, tc39 decorator implements is:
```ts
type Decorator = (input: Input, context: DecoratorContext) => Output | void
interface DecoratorContext {
  // xxx
}
```
link: https://github.com/tc39/proposal-decorators#2-calling-decorators

Which one is right? Should I use this babel implements?

**Comment by fedeci** [comment]
The correct one is definitely the TC39 one, the current Babel decorator proposal is outdated and has no longer been updated.

**Reply by XGHeaven** [reply]
Is there has a plan to update it?

**Reply by fedeci** [reply]
@nicolo-ribaudo is in contact with the proposal authors so he his more informed than me.

**Reply by nicolo-ribaudo** [reply]
Right, I believe @pzuraq is working on it!

**Reply by trusktr** [reply]
@pzuraq Is that true? Any status on it?

**Reply by pzuraq** [reply]
Yes it is, and the current status is I’m working on landing the `accessor` keyword at the moment. Goal is to have the transform finished by the next TC39 meeting.
